app/components/avo/views/resource_index_component.rb

```diff
@@ -45,18 +45,30 @@ def available_view_types
   end
 
   def can_create?
-    @resource.authorization.authorize_action(:create, raise_exception: false) && simple_relation?
+    @resource.authorization.authorize_action(:create, raise_exception: false) && simple_relation? && !is_field_read_only
   end
 
   def can_attach?
     klass = @reflection
     klass = @reflection.through_reflection if klass.is_a? ::ActiveRecord::Reflection::ThroughReflection
 
-    @reflection.present? && klass.is_a?(::ActiveRecord::Reflection::HasManyReflection)
+    @reflection.present? && klass.is_a?(::ActiveRecord::Reflection::HasManyReflection) && !is_field_read_only
   end
 
   def can_detach?
-    @reflection.present? && @reflection.is_a?(::ActiveRecord::Reflection::HasOneReflection) && @models.present?
+    @reflection.present? && @reflection.is_a?(::ActiveRecord::Reflection::HasOneReflection) && @models.present? && !is_field_read_only
+  end
+
+  def is_field_read_only
```

Comment:
In the context of `resource_index` there is no one `field`.

```suggestion
  def is_a_has_many_and_is_read_only
```